includes/install.php

```diff
@@ -417,3 +363,80 @@ function give_get_default_agreement_text() {
 
 	return apply_filters( 'give_get_default_agreement_text', $agreement, $org_name );
 }
+
+
+/**
+ * This function will install give related page which is not created already.
+ *
+ * @since 1.8.11
+ */
+function give_create_pages(){
+	// Bailout.
+	if( get_option( 'give_install_pages_created') ) {
+		return;
+	}
+
+	$options = array();
+
+	// Checks if the Success Page option exists AND that the page exists.
+	if ( ! get_post( give_get_option( 'success_page' ) ) ) {
+
+		// Donation Confirmation (Success) Page
+		$success = wp_insert_post(
+			array(
+				'post_title'     => esc_html__( 'Donation Confirmation', 'give' ),
+				'post_content'   => '[give_receipt]',
+				'post_status'    => 'publish',
+				'post_author'    => 1,
+				'post_type'      => 'page',
+				'comment_status' => 'closed'
+			)
+		);
+
+		// Store our page IDs
+		$options['success_page'] = $success;
+	}
+
+	// Checks if the Failure Page option exists AND that the page exists.
+	if ( ! get_post( give_get_option( 'failure_page' ) ) ) {
+
+		// Failed Donation Page
+		$failed = wp_insert_post(
+			array(
+				'post_title'     => esc_html__( 'Donation Failed', 'give' ),
+				'post_content'   => esc_html__( 'We\'re sorry, your donation failed to process. Please try again or contact site support.', 'give' ),
+				'post_status'    => 'publish',
+				'post_author'    => 1,
+				'post_type'      => 'page',
+				'comment_status' => 'closed'
+			)
+		);
+
+		$options['failure_page'] = $failed;
+	}
+
+	// Checks if the History Page option exists AND that the page exists.
+	if ( ! get_post( give_get_option( 'history_page' ) ) ) {
+		// Donation History Page
+		$history = wp_insert_post(
+			array(
+				'post_title'     => esc_html__( 'Donation History', 'give' ),
+				'post_content'   => '[donation_history]',
+				'post_status'    => 'publish',
+				'post_author'    => 1,
+				'post_type'      => 'page',
+				'comment_status' => 'closed'
+			)
+		);
+
+		$options['history_page'] = $history;
+	}
+
+	if( ! empty( $options ) ) {
+		$options = array_merge( give_get_settings(), $options );
+		give_update_option( 'give_settings', $options );
```

Comment:
give_update_option('give_settings', $options ); results in values incorrectly being added to the top level array 

![2017-07-11_13-25-32](https://user-images.githubusercontent.com/1571635/28089019-72004fda-663c-11e7-9539-4502968c8de2.png)